hack/generate-yaml.sh

```diff
@@ -0,0 +1,209 @@
+  -p    path to python or "docker" to use image (default "${PYTHON}")
+  -i    input directory (default ${TPL_DIR})
+  -o    output directory (default ${OUT_DIR})
+  -f    force overwrite of existing files
+  -h    prints this help screen
+EOF
+}
+
+while getopts ':fhi:o:k:p:' opt; do
+  case "${opt}" in
+  f)
+    OVERWRITE=1
+    ;;
+  h)
+    usage 1>&2; exit 1
+    ;;
+  o)
+    OUT_DIR="${OPTARG}"
+    ;;
+  i)
+    TPL_DIR="${OPTARG}"
+    ;;
+  k)
+    KUSTOMIZE="${OPTARG}"
+    [ "${KUSTOMIZE}" = "docker" ] || [ -f "${KUSTOMIZE}" ] || { echo "${KUSTOMIZE} does not exist" 1>&2; exit 1; }
+    ;;
+  p)
+    PYTHON="${OPTARG}"
+    [ "${PYTHON}" = "docker" ] || [ -f "${PYTHON}" ] || { echo "${PYTHON} does not exist" 1>&2; exit 1; }
+    ;;
+  \?)
+    { echo "invalid option: -${OPTARG}"; usage; } 1>&2; exit 1
+    ;;
+  :)
+    echo "option -${OPTARG} requires an argument" 1>&2; exit 1
+    ;;
+  esac
+done
+
+mkdir -p "${OUT_DIR}"
+
+ok_file() {
+  [ -f "${1}" ] || { echo "${1} is missing" 1>&2; exit 1; }
+}
+
+no_file() {
+  [ ! -f "${1}" ] || { echo "${1} already exists, overwrite with -f" 1>&2; exit 1; }
+}
+
+for f in ADDON CLUSTER MACHINES MACHINESET; do
+  eval "ok_file \"\${${f}_TPL_FILE}\""
+  [ -n "${OVERWRITE}" ] || eval "no_file \"\${${f}_OUT_FILE}\""
+done
+
+require_if_defined() {
+  while [ "${#}" -gt "0" ]; do
+    eval "[ ! \"\${${1}+x}\" ] || ${1}=\"\${${1}:?required if defined}\""
+    shift
+  done
+}
+
+require_if_defined CAPV_MANAGER_IMAGE \
+                   VSPHERE_DATACENTER \
+                   VSPHERE_DATASTORE \
+                   VSPHERE_RESOURCE_POOL \
+                   VSPHERE_FOLDER \
+                   VSPHERE_TEMPLATE
+
+# All variables used for yaml generation
+EXPORTED_ENV_VARS=
+record_and_export() {
+  eval "EXPORTED_ENV_VARS=\"${EXPORTED_ENV_VARS} -e ${1}\"; \
+        export ${1}=\"\${${1}${2}}\""
+}
+record_and_export CLUSTER_NAME          ':-capv-mgmt-example'
+record_and_export SERVICE_CIDR          ':-100.64.0.0/13'
+record_and_export CLUSTER_CIDR          ':-100.96.0.0/11'
+record_and_export CAPV_MANAGER_IMAGE    ':-'
+record_and_export VSPHERE_USER          ':?required'
+record_and_export VSPHERE_PASSWORD      ':?required'
+record_and_export VSPHERE_SERVER        ':?required'
+record_and_export VSPHERE_DATACENTER    ':-'
+record_and_export VSPHERE_DATASTORE     ':-'
+record_and_export VSPHERE_NETWORK       ':?required'
+record_and_export VSPHERE_RESOURCE_POOL ':-'
+record_and_export VSPHERE_FOLDER        ':-'
+record_and_export VSPHERE_TEMPLATE      ':-'
+
+verify_cpu_mem_dsk() {
+  eval "[[ \${${1}-} =~ [[:digit:]]+ ]] || ${1}=\"${2}\"; \
+        [ \"\${${1}}\" -ge \"${2}\" ] || { echo \"${1} must be >= ${2}\" 1>&2; exit 1; }; \
+        record_and_export ${1} ':-\"\${${1}}\"'"
+}
+verify_cpu_mem_dsk VSPHERE_NUM_CPUS 2
+verify_cpu_mem_dsk VSPHERE_MEM_MIB  2048
+verify_cpu_mem_dsk VSPHERE_DISK_GIB 20
+
+# TODO: check if KUBERNETES_VERSION has format "v1.13.6" and
```

Comment:
Looks like we can remove this comment now?

Reply:
No, because we still don't check for the leading `v`. We need to remove it, I just didn't bother to do that in the code yet.